server/service/hosts.go

```diff
@@ -735,12 +741,20 @@ func (svc *Service) MacadminsData(ctx context.Context, id uint) (*fleet.Macadmin
 		enrollmentStatus = "Enrolled (automated)"
 	}
 
-	data := &fleet.MacadminsData{
-		Munki: fleet.HostMunkiInfo{Version: version},
-		MDM: fleet.HostMDM{
+	if err == nil {
+		mdm = &fleet.HostMDM{
 			EnrollmentStatus: enrollmentStatus,
 			ServerURL:        serverURL,
-		},
+		}
+	}
+
+	if munkiInfo == nil && mdm == nil {
+		return nil, nil
```

Comment:
Curious why, will the UI treat this case differently? (If the UI expect this when both are not configured, then let's document it here).

Reply:
@lukeheath ^

Reply:
@lucasmrod Previously it was returning an object with all properties present, but empty strings. That makes it kind of clumsy for us to check on the frontend, because string is technically a value. Returning null allows us to use a cleaner `macadmins && ...` pattern.

Reply:
To clarify, if the host details response indicates whether macadmins is present or not we won't hit this endpoint for anything except valid hosts. However, a response of "null" more clearly indicates no data vs. an empty string with no indication.